.github/sync_changed_templates_with_cloud.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# shellcheck disable=SC2153
+main() {
+  local wiremock_cloud_url="$1"
+  local wiremock_cloud_api_token="$2"
+  cd /"$HOME"/wiremock/api-template-library
+  basic_api_template_file_path=/$HOME/wiremock/api-template-library/api/
+  modified_api_templates=$(git diff --relative=api --diff-filter=M)
+  deleted_api_templates=$(git diff --relative=api --diff-filter=D)
+  if [[ $modified_api_templates != '' ]]; then
+    (update_all_changed_api_templates "$basic_api_template_file_path" "$wiremock_cloud_url" "$wiremock_cloud_api_token")
+  fi
+
+  if [[ $deleted_api_templates != '' ]]; then
+    (delete_all_api_templates "$basic_api_template_file_path" "$wiremock_cloud_url" "$wiremock_cloud_api_token")
+  fi
+}
+
+update_all_changed_api_templates() {
+  basic_api_template_file_path=$1
+  wiremock_cloud_url=$2
+  wiremock_cloud_api_token=$3
+  api_changes_in_current_branch=$(git diff --relative=api --diff-filter=M --name-only main)
+  while IFS='' read -r changed_file; do
+    (update_api_template "$basic_api_template_file_path$changed_file" "$wiremock_cloud_url" "$wiremock_cloud_api_token")
+  done <<< "$api_changes_in_current_branch"
+}
+
+delete_all_api_templates() {
+  basic_api_template_file_path=$1
+  wiremock_cloud_url=$2
+  wiremock_cloud_api_token=$3
+  api_templates_deleted_in_current_branch=$(git diff --relative=api --diff-filter=D --name-only main)
```

Comment:
I think this will find any deleted file - but if, for instance, we change the logo from an svg to a gif that would be a deleted file. We need to check that it is the metadata file that is being deleted, I think